pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,298 @@
+			Name: "Deploying backup entry",
+			Fn:   flow.TaskFn(botanist.DeployBackupEntryInGarden),
+		})
+		wailtUntilBackupEntryInGardenReconciled = g.Add(flow.Task{
+			Name:         "Waiting until the backup entry has been reconciled",
+			Fn:           flow.TaskFn(botanist.WaitUntilBackupEntryInGardenReconciled),
+			Dependencies: flow.NewTaskIDs(deployBackupEntryInGarden),
+		})
+		deployETCD = g.Add(flow.Task{
+			Name:         "Deploying main and events etcd",
+			Fn:           flow.TaskFn(botanist.DeployETCD).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deploySecrets, wailtUntilBackupEntryInGardenReconciled),
+		})
+		waitUntilEtcdReady = g.Add(flow.Task{
+			Name:         "Waiting until main and event etcd report readiness",
+			Fn:           flow.TaskFn(botanist.WaitUntilEtcdReady),
+			Dependencies: flow.NewTaskIDs(deployETCD),
+		})
+		prepareControlPlaneDeployments = g.Add(flow.Task{
+			Name:         "Stop Kubernetes API Server in Shoot's namespace",
+			Fn:           flow.TaskFn(botanist.PrepareControlPlaneDeployments).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(shootHibernated),
+			Dependencies: flow.NewTaskIDs(waitUntilEtcdReady),
+		})
+		waitUntilAPIServerIsDown = g.Add(flow.Task{
+			Name:         "Wait for Kubernetes API Server to be scaled to zero",
+			Fn:           flow.TaskFn(botanist.WaitUntilKubeAPIServerScalesDown).SkipIf(shootHibernated),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying ingress DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyIngressDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying external domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyExternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying internal domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyInternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		createETCDSnapshot = g.Add(flow.Task{
+			Name:         "Create ETCD Snapshot",
+			Fn:           botanist.CreateETCDSnapshot,
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments, waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Scale ETCD to zero",
+			Fn:           botanist.ScaleETCDStatefulSetsToZero,
+			Dependencies: flow.NewTaskIDs(createETCDSnapshot),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Annotate Extensions with operation - migration",
+			Fn:           botanist.AnnotateExtensionCRDsForMigration,
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Annotate the shoot is eligibe for migration",
+			Fn:           botanist.AnnotateShootForMigration,
```

Comment:
@mandelsoft suggested to not use an annotation to trigger the restoration process in the new seed. Instead, just remove the `.status.seedName` in order to indicate that the shoot is ready for restoration. Then no annotation is needed. 
The gardenlet in the new seed should wait until `.status.seedName=nil` before it starts. WDYT?

Reply:
Another sync with @mandelsoft lead to the following proposal:
1. The `.status.seedName` is now set only after the reconciliation was successful - this should be changed in a way that the `.status.seedName` is set right at the beginning when the shoot gets reconciled/restored.
1. When the gardenlet responsible for the source seed has finished the migration preparation it sets `.status.lastOperation.type=Migrate` and `.status.lastOperation.state=Succeeded`, plus it sets `.status.seedName=nil`.
1. This will trigger the gardenlet responsible for the new target seed to start the restoration process. It will set `.status.seedName=<new-seed>` and `.status.lastOperation.type=Restore` and `.status.lastOperation.state=Processing`.
1. After a successful restoration it sets `.status.lastOperation.state=Succeeded`. Probably it makes sense to trigger a reconciliation flow now.

This should solve the problem that you mentioned that the gardenlet cannot distinguish the create/reconcile and restore cases. Now it can:
1. A create/reconciliation happens if `.status.lastOperation.type={Create,Reconcile}`
1. A restoration happens if `.status.lastOperation.type=Restore` and `.status.lastOperation.state!=Succeeded`.

If it crashes during the restoration and comes up again it can still see that it was previously doing the restoration and can start to do it again.
WDYT?

Reply:
Sounds promising, I will try to do it that way.

Reply:
I've implemented it that way, but I guess this has a certain little flaw that update of the Shoot's status didn't enqueue the object for reconciliation.